### PR TITLE
bugfix/utest_diagLevelTests

### PR DESCRIPTION
**Pull request by brettywhite**
- adding null pointer checks as the tests were failing when null

### CODE REVIEW

File: sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java [hunk 1]
```diff
@@ -14,29 +14,38 @@
...
+			for (int i = 0; i < levels.length; i++) {
+				levels[i] = thisDetail; //
+			}
 		}
 	}
 	
     public static void setLevel(Mod thisMod, DetailLevel thisDetail) {
-    	levels[thisMod.ordinal()] = thisDetail;
+		if (thisMod != null && thisDetail != null) {
+			levels[thisMod.ordinal()] = thisDetail;
```
Comment (askirk) [inline sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java]: Should there be another check for `.ordinal != null`?

File: sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java [hunk 2]
```diff
@@ -14,29 +14,38 @@
...
+		}
     }
 	
 	public static DetailLevel getLevel(Mod thisMod) {
-		return levels[thisMod.ordinal()];
+		if (thisMod != null) {
+			return levels[thisMod.ordinal()];
```
Comment (askirk) [inline sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java]: Same thing

File: sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java (hunk 1 above)
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/trace/DiagLevel.java]: If mod is not null, it is guaranteed to have an ordinal value.
